src/Raven.Client/Documents/Operations/Backups/BackupStatus.cs

```diff
@@ -138,13 +137,20 @@ public UploadProgress()
 
         public double BytesPutsPerSec { get; set; }
 
-        public long UploadTimeInMs => _sw.ElapsedMilliseconds;
+        public long UploadTimeInMs => _sw?.ElapsedMilliseconds ?? 0;
 
         public void ChangeState(UploadState newState)
         {
             UploadState = newState;
-            if (newState == UploadState.Done)
-                _sw.Stop();
+            switch (newState)
+            {
+                case UploadState.PendingUpload:
+                    _sw = Stopwatch.StartNew();
+                    break;
+                case UploadState.Done:
+                    _sw.Stop();
```

Comment:
Can this be null?
Can we have the install at all times and just call Restart on PendingUpdate?

Reply:
Shouldn't be since we set UploadState to PendingUpload before starting the updload

Reply:
I'm not sure we can always ensure that this happens.
Better to avoid the potential NRE

Reply:
Done.